File: Level 3/입국심사/solution.js
function solution(n, times) {
  let low = 1;
  let high = n * Math.max(...times);

  while (low <= high) {
    const mid = Math.floor((low + high) / 2);
    const people = times.reduce((acc, time) => acc + Math.floor(mid / time), 0);

    if (people < n) {
      low = mid + 1;
    } else {
      high = mid - 1;
    }
  }

  return low;
}

File: Level 3/입국심사/solution.py
def solution(n, times):
    low = 1
    high = n * max(times)

    while low <= high:
        mid = (low + high) // 2
        count = sum(mid // time for time in times)

        if count < n:
            low = mid + 1
        else:
            high = mid - 1

    return low
